onnxruntime/test/providers/qnn/resize_test.cc

```diff
@@ -286,48 +299,103 @@ TEST_F(QnnCPUBackendTests, Resize2xLinearAlignCorners_scales) {
+  std::vector<float> input_data = GetFloatDataInRange(-10.0f, 10.0f, 48);
+  RunQDQResizeOpTest<uint8_t>(TestInputDef<float>({1, 3, 4, 4}, false, input_data),
+                              {1, 3, 8, 8}, "nearest", "half_pixel", "round_prefer_floor",
+                              ExpectedEPNodeAssignment::All);
+}
+
+// Test that the nearest_mode "ceil" is not supported on the HTP backend.
+TEST_F(QnnHTPBackendTests, ResizeU8_NearestModeCeil_Unsupported) {
+  std::vector<float> input_data = GetFloatDataInRange(-10.0f, 10.0f, 48);
+  RunQDQResizeOpTest<uint8_t>(TestInputDef<float>({1, 3, 4, 4}, false, input_data),
+                              {1, 3, 8, 8}, "nearest", "asymmetric", "ceil",
+                              ExpectedEPNodeAssignment::None);
+}
+
+// Test 3x QDQ Resize mode: "nearest", coordinate_transformation_mode: "asymmetric", nearest_mode: "floor".
+// QNN EP uses QNN's ResizeNearestNeighbor op.
 TEST_F(QnnHTPBackendTests, ResizeU8_3xNearestAsymmetricFloor) {
-  RunQDQResizeOpTest<uint8_t>(TestInputDef<float>({1, 3, 4, 4}, false, -10.0f, 10.0f),
+  std::vector<float> input_data = GetFloatDataInRange(-10.0f, 10.0f, 48);
+  RunQDQResizeOpTest<uint8_t>(TestInputDef<float>({1, 3, 4, 4}, false, input_data),
                               {1, 3, 12, 12}, "nearest", "asymmetric", "floor",
                               ExpectedEPNodeAssignment::All);
 }
 
+// Test 2x QDQ Resize mode: "nearest", coordinate_transformation_mode: "asymmetric", nearest_mode: "round_prefer_floor"
+// QNN EP uses QNN's Resize op.
+TEST_F(QnnHTPBackendTests, ResizeU8_2xNearestAsymmetricRoundPreferFloor) {
+  std::vector<float> input_data = GetFloatDataInRange(-10.0f, 10.0f, 8);
+  RunQDQResizeOpTest<uint8_t>(TestInputDef<float>({1, 2, 2, 2}, false, input_data),
+                              {1, 2, 4, 4}, "nearest", "asymmetric", "round_prefer_floor",
+                              ExpectedEPNodeAssignment::All);
+}
+
+// Test 3x QDQ Resize mode: "nearest", coordinate_transformation_mode: "asymmetric", nearest_mode: "round_prefer_floor"
+// QNN EP uses QNN's Resize op.
+//
+// TODO: Inaccuracy detected for output 'output_0', element 2.
+// Output quant params: scale=0.078431375324726105, zero_point=127.
+// Expected val: -3.3333334922790527
+// QNN QDQ val: -9.960784912109375 (err 6.6274514198303223)
+// CPU QDQ val: -3.2941176891326904 (err 0.039215803146362305)
+//
+// More debugging info:
+// Input elements f32[1,1,2,2] = -10.0000000 -3.33333349 3.33333302 10.0000000
+// ORT CPU EP (f32 model) outputs: -10.0000000 -10.0000000 -3.33333349 -3.33333349 -3.33333349 -3.33333349 -10.00 ...
+// ORT CPU EP (qdq model) outputs: -9.96078491 -9.96078491 -3.29411769 -3.29411769 -3.29411769 -3.29411769 -9.961 ...
+// ORT QNN EP (qdq model) outputs: -9.96078491 -9.96078491 -9.96078491 -3.37254906 -3.37254906 -3.37254906 -9.961 ...
+TEST_F(QnnHTPBackendTests, DISABLED_ResizeU8_3xNearestAsymmetricRoundPreferFloor) {
```

Comment:
We did not previously support this combination of parameters. But I decided to test it now that we have many fixes. Will report to QC.